File: algorithms/gr-bfsrh/bfs.cs
// C# program to print BFS traversal
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

// This class represents a directed
// graph using adjacency list
// representation
class Graph{
	
private int _V;

LinkedList<int>[] _adj;

public Graph(int V)
{
	_adj = new LinkedList<int>[V];
	for(int i = 0; i < _adj.Length; i++)
	{
		_adj[i] = new LinkedList<int>();
	}
	_V = V;
}

public void AddEdge(int v, int w)
{		
	_adj[v].AddLast(w);

}

public void BFS(int s)
{
	bool[] visited = new bool[_V];
	for(int i = 0; i < _V; i++)
		visited[i] = false;
	
  LinkedList<int> queue = new LinkedList<int>();
	
	visited[s] = true;
	queue.AddLast(s);		

	while(queue.Any())
	{
		s = queue.First();
		Console.Write(s + " " );
		queue.RemoveFirst();

		LinkedList<int> list = _adj[s];

		foreach (var val in list)			
		{
			if (!visited[val])
			{
				visited[val] = true;
				queue.AddLast(val);
			}
		}
	}
}

static void Main(string[] args)
{
	Graph g = new Graph(4);
	
	g.AddEdge(0, 1);
	g.AddEdge(0, 2);
	g.AddEdge(1, 2);
	g.AddEdge(2, 0);
	g.AddEdge(2, 3);
	g.AddEdge(3, 3);
	
	Console.Write("Following is Breadth First " +
				"Traversal(starting from " +
				"vertex 2)\n");
	g.BFS(2);
}
}

File: algorithms/gr-bfsrh/bfs.js
// function to add edge to graph
function addedge(u, v, adj){
    adj[u].push(v);
    adj[v].push(u);
}

// function to print BFS traversal from a given source s
function bfs(s, adj){
    let visited = new Array(adj.length).fill(false);
    let q = [];
    q.push(s);
    visited[s] = true;
    for(let i=0;i<adj.length;i++){
        if(visited[i]==false){
            while(q.length > 0){
                let u = q.shift();
                console.log(u);
                for(let i = 0; i < adj[u].length; i++){
                    if(visited[adj[u][i]] == false){
                        q.push(adj[u][i]);
                        visited[adj[u][i]] = true;
                    }
                }
            }
        }
    }
}

// Driver code
let adj = new Array(14).fill(new Array());

addedge(0, 1, adj);
addedge(0, 2, adj);
addedge(0, 3, adj);
addedge(1, 4, adj);
addedge(1, 5, adj);
addedge(2, 6, adj);
addedge(2, 7, adj);
addedge(3, 8, adj);
addedge(8, 9, adj);
addedge(8, 10, adj);
addedge(9, 11, adj);
addedge(9, 12, adj);
addedge(9, 13, adj);

let start = 0;
bfs(start, adj);

